Dockerfile running for frontend and Secrets for Terraform

Raymondtaoo opened the pull request:
dockerfile for frontend builds and runs

Added GitHub Actions Workflow for terraform configurations using secrets as environment variable and changed credentials so that it's not hardcoded

rawanmahdi reviewed (commented):
why did you remove the terraform credentials setting block?
`   - name: Set up Terraform credentials
        run: |
          echo "GOOGLE_CREDENTIALS=${{ secrets.DEFAULT_SA_KEY }}" >> $GITHUB_ENV
`

Raymondtaoo commented:
> why did you remove the terraform credentials setting block? ` - name: Set up Terraform credentials run: | echo "GOOGLE_CREDENTIALS=${{ secrets.DEFAULT_SA_KEY }}" >> $GITHUB_ENV`

whoops didn't mean to, i forgot to pull before pushing last merge so I had merge conflicts, I fixed it and added hold for approval into the yml

ausbennett reviewed (approved):
works, and looks good! Concerned about image size, ~700mb when I build it locally. But we will worry about that when we have to : )
